### PR TITLE
Add overall score to the redux state

### PR DESCRIPTION
**Pull request by danielFangstrom**
## Summary

This PR can be summarized in the following changelog entry:

* Adds the overall score for each type of analysis to the redux state

## Relevant technical choices:

* Adds actions and reducers for overall scores to the content analysis reducer

## Test instructions

This PR can be tested by following these steps:

* Run `yarn install` and `yarn test`, and check whether the tests pass
* You can also expose the store and the action to the window in edit.js, and dispatch it (store.dispatch( action() ) and check whether the state was changed correctly regarding the overall scores. Readability results and SEO results should have separate overall scores, and SEO should also have an overall score for each keyword.

Fixes https://github.com/Yoast/wordpress-seo/issues/9308

### CODE REVIEW

File: composites/Plugin/ContentAnalysis/reducers/contentAnalysis/overallScoreReducer.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * Sets the overall score for SEO results for one or more keywords.
+ *
+ * @param {Object} action The action.
+ * @returns {Object} The overall score per keyword.
+ */
+function setOverallScoreSeo( action ) {
+	let scorePerKeyword = {};
+	action.scorePerKeyword.forEach( function( keywordResultsPair ) {
```
Comment (maartenleenders) [inline composites/Plugin/ContentAnalysis/reducers/contentAnalysis/overallScoreReducer.js]: I'd probably call `setOverallScoreSeo( action.scorePerKeyword )` elsewhere, rather than making this function work on action objects. Small difference though.

File: composites/Plugin/ContentAnalysis/actions/tests/ContentAnalysisTest.js [hunk 2]
```diff
@@ -128,3 +132,53 @@ describe( "removeKeyword function", function() {
...
+			results: results,
+		};
+		const actual = setOverallScoreReadability( results );
+		expect( actual ).toEqual( expected );
+	} );
+} );
+
+describe( "setOverallScore seo function", function() {
+	it( "creates the setOverallScore for seo results action", function() {
+		let keyword1 = "keyword1";
```
Comment (maartenleenders) [inline composites/Plugin/ContentAnalysis/actions/tests/ContentAnalysisTest.js]: Seems unnecessary to explicitly assign this variable; you can hard-code the string in the mock scorePerKeyword object.

File: composites/Plugin/ContentAnalysis/reducers/tests/overallScoreReducerTest.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
+import { SET_OVERALL_SCORE_SEO, SET_OVERALL_SCORE_READABILITY } from "../../actions/contentAnalysis";
+import { overallScoreReducer } from "../contentAnalysis/overallScoreReducer";
+
+describe( "SET_OVERALL_SCORE for seo action", () => {
+	it( "sets overall score for an seo analysis result", () => {
+		const state = {};
+		let firstKeyword = "thisIsMyFirstKeyword";
```
Comment (maartenleenders) [inline composites/Plugin/ContentAnalysis/reducers/tests/overallScoreReducerTest.js]: Same as above: I appreciate not wanting to hard-code in the object, but adding the `firstKeyword` and `"thisIsMyFirstKeyword"` complicates the CR at least 😜 . 

It's a difference in style I guess, I'll leave it up to you.

**Comment by maartenleenders** [comment]
CR 🚧 

Code's fine, but as discussed we'd like the Redux store to look as follows:
```
analysis: {
    keyword1: {
        results: [],
        overallScore: whatever format
    },
    keyword2: {
        results: [],
        overallScore: whatever format
    },
}
```

File: composites/Plugin/ContentAnalysis/actions/contentAnalysis.js [hunk 4]
```diff
@@ -119,3 +122,32 @@ export function updateReadabilityResult( result ) {
...
 }
+
+/**
+ * An action creator for setting the overall score for a readability result.
+ *
+ * @param {Object} results The overall score.
+ *
+ * @returns {Object} A set overall score action.
+ */
+export function setOverallScoreReadability( overallScore ) {
```
Comment (IreneStr) [inline composites/Plugin/ContentAnalysis/actions/contentAnalysis.js]: setOverallReadabilityScore is easier to read (also for the type below and the SEO ones)

File: composites/Plugin/ContentAnalysis/actions/tests/ContentAnalysisTest.js [hunk 5]
```diff
@@ -12,6 +12,10 @@ import {
 	removeKeyword,
 	setReadabilityResults,
 	setSeoResultsForKeyword,
+	SET_OVERALL_SCORE_READABILITY,
```
Comment (IreneStr) [inline composites/Plugin/ContentAnalysis/actions/tests/ContentAnalysisTest.js]: Please don't mix actions and action creators in the import

File: composites/Plugin/ContentAnalysis/reducers/contentAnalysis/readabilityResultsReducer.js [hunk 6]
```diff
@@ -40,14 +47,16 @@ function updateReadabilityResult( state, action ) {
  *
  * @param {Object} state The current state of the object.
  * @param {Object} action The current action received.
- * @returns {Object} The updated readability results object.
+ * @returns {Object} The updated readability rewsults object.
```
Comment (IreneStr) [inline composites/Plugin/ContentAnalysis/reducers/contentAnalysis/readabilityResultsReducer.js]: Where did this 'w' come from? ;)

**Comment by danielFangstrom** [comment]
I've done overall changes to the reducers in order to deal with the changed structure of the results in the state. The 'updateReadabilityResult' function in readabilityResultsReducer looks a bit clunky now, so it might need to be separated into different functions. However, what it did before already surpassed simply updating results and probably should have been separated for some time.

File: composites/Plugin/ContentAnalysis/reducers/contentAnalysis/overallScoreReducer.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+export function setOverallReadabilityScore( state, action ) {
+	return Object.assign( {}, state,
+		{ overallScore: action.overallScore }
+	);
+}
+
+/**
+ * Sets the overall score for SEO results for a keyword.
+ *
+ * @param {Object} state The state.
```
Comment (igorschoester) [inline composites/Plugin/ContentAnalysis/reducers/contentAnalysis/overallScoreReducer.js]: Missing a space for the comment alignment.

File: composites/Plugin/ContentAnalysis/actions/contentAnalysis.js [hunk 8]
```diff
@@ -119,3 +122,33 @@ export function updateReadabilityResult( result ) {
...
+*
+* @param {Object} overallScore The overall score.
+* @param {Object} keyword The keyword the overall score is for.
+*
+* @returns {Object} A set overall score action.
+*/
+export function setOverallSeoScore( overallScore, keyword ) {
+	return {
+		type: SET_OVERALL_SEO_SCORE,
+		keyword: keyword,
```
[the remark this answers is not in the document]
Comment (danielFangstrom) [reply composites/Plugin/ContentAnalysis/actions/contentAnalysis.js]: Good idea, but I think for now we can keep the current syntax. Might be nice to change in the future though, particularly if we add more actions.

**Comment by IreneStr** [comment]
I've removed the unused `overallScoreReducer `, and moved `setOverallSeoScore` and `setOverallReadabilityScore` to the `keywordResultsReducer.js ` and the `readabilityResultsReducer.js` respectively. 

When you run the tests, some checkboxtests are failing. This is also the case on `feature/content-analysis-in-sidebar`, which means the failing tests are not part of this issue/PR. I'll look into this later.

File: webpack.config.js [hunk 9]
```diff
@@ -54,6 +54,7 @@ module.exports = {
 							development: {
 								plugins: [
 									"react-hot-loader/babel",
+									"babel-plugin-transform-object-rest-spread",
```
Comment (atimmer) [inline webpack.config.js]: This should not be necessary, why did you add it?

**Comment by igorschoester** [comment]
CR & ACC 👍
